OpenDreamServer/Dream/Procs/ExecutionContext.cs

```diff
@@ -0,0 +1,165 @@
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Text;
+using OpenDreamServer.Dream.Objects;
+using OpenDreamShared.Dream.Procs;
+
+namespace OpenDreamServer.Dream.Procs {
+    enum ProcStatus {
+        Returned,
+        Deferred,
+        Called,
+    }
+
+    abstract class DreamProc {
+        public string Name { get; }
+
+        // This is currently publically settable because the loading code doesn't know what our super is until after we are instantiated

```

Comment:
```suggestion
        // This is currently publicly settable because the loading code doesn't know what our super is until after we are instantiated
```

:^)